Patch to avoid upstream ResourceServer panic

mikhailshilkov opened the pull request:
Resolve https://github.com/pulumi/pulumi-auth0/issues/285 by patching the check upsteam
It's my first patch ever, so let me know if I'm doing it wrong.

iwahbe commented on provider/resources.go:
```diff
@@ -52,14 +52,15 @@ func Provider() tfbridge.ProviderInfo {
+		Homepage:         "https://pulumi.io",
+		GitHubOrg:        "auth0",
+		Repository:       "https://github.com/pulumi/pulumi-auth0",
+		UpstreamRepoPath: "./upstream",
```
Unrelated, but we should really set `Version: version.version,` here.
```suggestion
		UpstreamRepoPath: "./upstream",
		Version:          version.Version,
```

mikhailshilkov replied:
Added

iwahbe commented on provider/go.mod:
```diff
@@ -3,6 +3,7 @@ module github.com/pulumi/pulumi-auth0/provider/v2
 
 replace (
+	github.com/auth0/terraform-provider-auth0 => ../upstream
 	github.com/auth0/terraform-provider-auth0/shim => ./shim
```
We don't want to mix a shim with an upstream, since it adds upgrade complexity. Can we move the shim into `upstream/shim` via another patch?

EronWright replied:
@iwahbe could you explain a bit more, I'd like to better understand this principle.  thx

iwahbe replied:
@EronWright we use a `provider/shim/` module to get access to functions behind `internal` in the upstream provider. This means that we have an additional `go.mod` that we need to update and keep in sync. We reduce the number of go modules by moving the shim into the `upstream` with a patch.

This reduces the number of places where we need to update a `replace` for testing and building the provider.

iwahbe replied:
It also has the nice benefit of ensuring that we are compiling against the patched upstream, since compiling against the raw provider will error: the raw provider doesn't expose a `shim` package.

iwahbe reviewed (commented):
The patch looks good to me. Let's make sure we have a test, both to confirm that the patch works and to make it easy to check when the patch is safe to remove.

mikhailshilkov commented:
I think I addressed review suggestions - thank you all. I'm not sure why lots of metadata changed - it doesn't look related but I got it from `make build`... Any ideas?

EronWright reviewed (commented):
LGTM except for the metadata stuff, seems to genuinely be losing information.

iwahbe reviewed (approved):
LGTM.

The metadata stuff is just discarding empty objects. We want to include that.
